doc/user_guide/type-system.md

```diff
@@ -0,0 +1,48 @@
+Type system
+===========
+
+Simple unified access to all data types
+---------------------------------------
+The datamodel API for accessing properties of an instance in SOFT, has
+separate getters and setters for each type and number of dimensions.
+DLite generalize and simplifies this by describing types and
+dimensionality of properties with 4 parameters:
+
+  - `type`: an enum defining the type of the data item (or items if
+    it has dimensions), e.g. whether it is an integer, float or string...
+    The table below summarises the implemented dtype's.
+  - `size`: the size of a data item in bytes.
+  - `ndims`: number of dimensions.  Scalars has ``ndims=0``.
+  - `dims`: array of length `ndims` with the length of each dimension.
+
+By taking these parameters as arguments, the functions
+DLite_datamodel_get_property() and DLite_datamodel_set_property() can handle
```

Comment:
What about [dlite_datamodel_get_property()](https://github.com/SINTEF/dlite/blob/master/src/dlite-datamodel.c#L118) and [dlite_datamodel_set_property()](https://github.com/SINTEF/dlite/blob/master/src/dlite-datamodel.c#L134)

(Or the corresponding link to the definitions)

Reply:
definitely, these were very old names